torch/_meta_registrations.py

```diff
@@ -3166,6 +3099,19 @@ def meta__foreach_copy_inplace(self, src, non_blocking=False):
     _check_foreach_binop_tensor_lists(self, src)
 
 
+@register_meta([aten._local_scalar_dense])
+def meta__local_scalar_dense(self):
+    torch._check(
+        isinstance(self, Tensor),
+        lambda: f"The first argument must be Tensor, but got {type(self)}.",
+    )
+    torch._check(
+        self.dim() == 0 and self.numel() == 1,
+        lambda: f"The first argument must be 0-dim, but got {self.dim()}-dim.",
+    )
+    return torch.zeros(size=(1,), dtype=self.dtype).item()
```

Comment:
This is not correct. This function is data-dependent, as you found when you tried to write `self.item()` and the tests yelled at you. The issue is that meta functions work on FakeTensors, which are tensors with no data. As such, you cannot call `.item()` on them.

When we are working with dynamo, we sometimes can just pull out a symbol of the correct dtype (SymInt/SymBool/SymFloat) and continue tracing using it. This is done in https://github.com/pytorch/pytorch/blob/f89291c1eb2e0d8768d9eb3d326033ae95d43ce7/torch/_subclasses/fake_tensor.py#L562-L574.

Now, the thing here is that there is no reason why any of these functions should have any data dependent behaviour. IIRC, you just hit this issue in `add` and a couple others, right? Let's then have 2 separate PRs, one with all the functions and a second one looking at these problematic ones.

Reply:
I've removed the implementation of `local_scalar_dense` from this PR and will work on an alternative.